src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListener.cs

```diff
@@ -26,16 +27,16 @@ internal class HttpDiagnosticListener : IDiagnosticListener
 		internal readonly ConcurrentDictionary<HttpRequestMessage, Span> ProcessingRequests = new ConcurrentDictionary<HttpRequestMessage, Span>();
 
 		public HttpDiagnosticListener(IApmAgent components) =>
-			(Logger, ConfigurationReader) = (components.Logger?.Scoped(nameof(HttpDiagnosticListener)), components.ConfigurationReader);
+			(_logger, _configurationReader) = (components.Logger?.Scoped(nameof(HttpDiagnosticListener)), components.ConfigurationReader);
```

Comment:
Well, I'm ok to discuss this, but I'd suggest moving this outside this PR, mainly because this syntax is already used in multiple places, so either we collectively not use it (but I'd push it separately due to lots of unrelated change lines), or we keep it. Also, this was already like that before the PR. I personally like this, and I think I saw @Mpdreamz also doing this.

Reply:
In many cases destructering assignments can be handy and easier to read, but if the right side of the assignment does too much it can be best to move it to regular multiline assigns :) 

In this case I'd personally not opt for multiple lines but it's a close call. 

Per case judgements seems best to me rather than blanket rules on new syntax usage :)